libraries/bot-azure/src/test/java/com/microsoft/bot/azure/TranscriptStoreTests.java

```diff
@@ -0,0 +1,589 @@
+                // all good
+            }
+        }
+    }
+
+    @Test
+    public void logActivities() {
+        if (EMULATOR_IS_RUNNING) {
+            ConversationReference conversation = TestAdapter
+                .createConversationReference(UUID.randomUUID().toString(), "User1", "Bot");
+            TestAdapter adapter = new TestAdapter(conversation).use(new TranscriptLoggerMiddleware(transcriptStore));
+            new TestFlow(adapter, turnContext -> {
+                Activity typingActivity = new Activity() {
+                    {
+                        setType(ActivityTypes.TYPING);
+                        setRelatesTo(turnContext.getActivity().getRelatesTo());
+                    }
+                };
+                turnContext.sendActivity(typingActivity).join();
+                try {
+                    TimeUnit.SECONDS.sleep(5);
+                } catch (InterruptedException e) {
+                    // Empty error
+                }
+                turnContext.sendActivity(String.format("echo:%s", turnContext.getActivity().getText())).join();
+                return CompletableFuture.completedFuture(null);
+            })
+                .send("foo").assertReply(activity -> {
+                    Assert.assertTrue(activity.isType(ActivityTypes.TYPING));
+            }).assertReply("echo:foo")
+                .send("bar").assertReply(activity -> {
+                Assert.assertTrue(activity.isType(ActivityTypes.TYPING));
+            }).assertReply("echo:bar")
+                .startTest().join();
+
+            PagedResult<Activity> pagedResult = null;
+            try {
+                pagedResult = this.getPagedResult(conversation, 6, null).join();
+            } catch (TimeoutException ex) {
+                Assert.fail();
+            }
+            Assert.assertEquals(6, pagedResult.getItems().size());
+            if(pagedResult.getItems().get(0).isType(ActivityTypes.MESSAGE)) {
+                Assert.assertEquals("foo", pagedResult.getItems().get(0).getText());
+            } else {
+                Assert.fail();
+            }
+
+            Assert.assertTrue(pagedResult.getItems().get(1) != null);
+            if(!pagedResult.getItems().get(1).isType(ActivityTypes.TYPING)) {
+                Assert.fail();
+            }
+
+            if(pagedResult.getItems().get(2).isType(ActivityTypes.MESSAGE)) {
+                Assert.assertEquals("echo:foo", pagedResult.getItems().get(2).getText());
+            } else {
+                Assert.fail();
+            }
+
+            if(pagedResult.getItems().get(3).isType(ActivityTypes.MESSAGE)) {
+                Assert.assertEquals("bar", pagedResult.getItems().get(3).getText());
+            } else {
+                Assert.fail();
+            }
+
+            Assert.assertTrue(pagedResult.getItems().get(4) != null);
+            if(!pagedResult.getItems().get(4).isType(ActivityTypes.TYPING)) {
+                Assert.fail();
+            }
```

Comment:
Remove this, maybe we can add a new assert to check this

`Assert.assertTrue(... .isType(ActivityTypes.TYPING))`